service/controller/resource/tccp/create.go

```diff
@@ -237,52 +240,714 @@ func (r *Resource) getCloudFormationTags(cr v1alpha1.Cluster) []*cloudformation.
 	return awstags.NewCloudFormation(tags)
 }
 
-func (r *Resource) newTemplateParams(ctx context.Context, cr v1alpha1.Cluster, t time.Time) (adapter.Adapter, error) {
+func newIAMPoliciesParams(ctx context.Context, cr v1alpha1.Cluster) (*template.ParamsMainIAMPolicies, error) {
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
-		return adapter.Adapter{}, microerror.Mask(err)
+		return nil, microerror.Mask(err)
 	}
 
-	// XXX: DEBUG
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("tccp.newTemplateParams(): cc.Spec.TenantCluster.TCCP.AvailabilityZones: %#v", cc.Spec.TenantCluster.TCCP.AvailabilityZones))
+	var iamPolicies *template.ParamsMainIAMPolicies
+	{
+		iamPolicies = &template.ParamsMainIAMPolicies{
+			ClusterID:         key.ClusterID(&cr),
+			EC2ServiceDomain:  key.EC2ServiceDomain(cc.Status.TenantCluster.AWS.Region),
+			MasterPolicyName:  key.PolicyNameMaster(cr),
+			MasterProfileName: key.ProfileNameMaster(cr),
+			MasterRoleName:    key.RoleNameMaster(cr),
+			RegionARN:         key.RegionARN(cc.Status.TenantCluster.AWS.Region),
+			KMSKeyARN:         cc.Status.TenantCluster.Encryption.Key,
+			S3Bucket:          key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID),
+		}
+	}
+
+	return iamPolicies, nil
+}
+func newInternetGatewayParams(ctx context.Context, cr v1alpha1.Cluster) (*template.ParamsMainInternetGateway, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var internetGateways []template.ParamsMainInternetGatewayInternetGateway
+	for _, az := range cc.Spec.TenantCluster.TCCP.AvailabilityZones {
+		ig := template.ParamsMainInternetGatewayInternetGateway{
+			InternetGatewayRoute: key.SanitizeCFResourceName(key.PublicInternetGatewayRouteName(az.Name)),
+			RouteTable:           key.SanitizeCFResourceName(key.PublicRouteTableName(az.Name)),
+		}
+
+		internetGateways = append(internetGateways, ig)
+	}
+
+	var internetGateway *template.ParamsMainInternetGateway
+	{
+		internetGateway = &template.ParamsMainInternetGateway{
+			ClusterID:        key.ClusterID(&cr),
+			InternetGateways: internetGateways,
+		}
+	}
+
+	return internetGateway, nil
+}
+func newInstanceParams(ctx context.Context, cr v1alpha1.Cluster, t time.Time, r *Resource) (*template.ParamsMainInstance, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	c := template.SmallCloudconfigConfig{
+		S3URL: fmt.Sprintf("s3://%s/%s", key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID), key.S3ObjectPathTCCP(&cr)),
+	}
+	rendered, err := pkgtemplate.Render(key.CloudConfigSmallTemplates(), c)
```

Comment:
Rendering should not be done here. This here is only for injecting parameters. Check the other resources how they implement it. I am fine when you want to keep it easy in this PR and follow up separately.

Reply:
I would like to see a TODO issue for this. See https://github.com/giantswarm/fmt/blob/master/go/todo_comments.md.